src/util/Interrupt.cpp

```diff
@@ -37,16 +43,23 @@ Interrupt::request()
     requested = true;
 }
 
+void
+Interrupt::requestForCurrentThread()
+{
+    requested_for_thread = true;
+}
+
 void
 Interrupt::cancel()
 {
     requested = false;
+    requested_for_thread = false;
```

Comment:
`cancel()` is invoked from `GEOS_init_r()` — that doesn't seem quite right if we're using it to cancel an interrupt request for a specific context. Does cancellation even make sense for context-specific interruption?